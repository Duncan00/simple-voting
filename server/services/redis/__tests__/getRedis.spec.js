'use strict';

const on = jest.fn();
const ping = jest.fn();
const mockRedisClusterClass = function() {
	this.on = on;
	this.ping = ping;
};
jest.doMock('ioredis', () => ({
	Cluster: mockRedisClusterClass
}));
const getRedis = require('../getRedis');

describe('getRedis', () => {
	afterEach(() => {
		jest.resetModules();
		jest.resetAllMocks();
	});

	test('Setup successfully', async () => {
		ping.mockResolvedValue('PONG');
		const nodes = {};
		const logger = {
			info: jest.fn(),
			debug: jest.fn()
		};

		const redis = await getRedis({nodes, logger});

		expect(redis).toMatchSnapshot();
		expect(on).toHaveBeenCalledTimes(1);
		expect(ping).toHaveBeenCalledTimes(1);
	});

	test('Throw error if ping failed', async () => {
		ping.mockResolvedValue('Some Error');
		const nodes = {};
		const logger = {
			info: jest.fn(),
			debug: jest.fn()
		};

		let t;
		try {
			await getRedis({nodes, logger});
		} catch (e) {
			t = e;
		}

		expect(t).toMatchSnapshot();
		expect(on).toHaveBeenCalledTimes(1);
		expect(ping).toHaveBeenCalledTimes(1);
	});
});
