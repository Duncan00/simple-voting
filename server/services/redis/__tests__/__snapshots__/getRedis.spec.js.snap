// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRedis Setup successfully 1`] = `
mockRedisClusterClass {
  "on": [MockFunction] {
    "calls": Array [
      Array [
        "error",
        [Function],
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": undefined,
      },
    ],
  },
  "ping": [MockFunction] {
    "calls": Array [
      Array [],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": Promise {},
      },
    ],
  },
}
`;

exports[`getRedis Throw error if ping failed 1`] = `[Error: [REDIS] Health check failed: Some Error]`;
